app/controllers/kaui/admin_tenants_controller.rb

```diff
@@ -161,11 +161,11 @@ def new_plan_currency
     options[:api_key] = @tenant.api_key
     options[:api_secret] = @tenant.api_secret
 
-    catalogs = Kaui::Catalog::get_catalog_json(false, options)
+    catalog = Kaui::Catalog::get_catalog_json(true, options)
 
     # seek if plan id exists
-    catalogs.each do |catalog|
-      catalog[:plans].each { |plan| is_plan_id_found |= plan.plan_id == plan_id }
+    catalog.products.each do |product|
+      product.plans.each { |plan| is_plan_id_found |= plan.name == plan_id }
       break if is_plan_id_found
```

Comment:
Oh boy... I had to re-read the code, I couldn't believe our api was such that if we get the latest, we need to use `plan.name` and if not `plan.plan_id`, but [apparently this is the case](https://github.com/killbill/killbill-admin-ui/blob/work-for-release-0.19.x/app/models/kaui/catalog.rb#L8) 😢 .

Anyways, your change looks good.